### DISCUSSION TITLE
Accelerometer need to be reset on microbit every time or onGesture() will not work

### DISCUSSION THREAD

**Discussion by binnneal** [ARMmbed/DAPLink#1057] [General]
HI,

I am new to the forum and please guide me if the post belongs to somewhere else.

I have several microbits v2.21 running DAPLink Firmware v0257.  Recently 3 of the boards started to have issues with accelerometer functions that may or may not be firmware related.

Here is the Javascript to reproduce the problem:

```
input.onGesture(Gesture.Shake, function () {
    basic.showNumber(1)
})
// input.calibrateCompass()
basic.showNumber(0)
```

Expected behavior: LED shows "0" upon start and then changes to "1" after shaking the board.
Test result: LED shows "0" upon start and doesn't change after shaking the board.

However, if I uncomment the line "input.calibrateCompass()", I will be able to see the board responding to shakes as expected.  However, the problem recurs the next time if I skip the calibration.  Hence, I would have to recalibrate/reset the accelerometer every time for the sensor work as expected.

Interestingly, it only affects some boards but not all of them.  I tried different [beta] firmware but it didn't solve the problem.

Does anyone experience the same problem?  

[DETAILS.TXT](https://github.com/ARMmbed/DAPLink/files/13061698/DETAILS.TXT)

Many Thanks,

Yong

**Comment by mbrossard** [comment]
@microbit-carlos, would you know if this could be related to DAPLink?

**Comment by microbit-carlos** [comment]
Thanks @mbrossard, hi @binnneal !

The motion sensors wouldn't be affected by the DAPLink firmware (technically they share an interrupt signal, but this doesn't look related to interrupts).

If you could open a ticket in https://support.microbit.org we'll be able to help you better over there.

In the meantime, as this will be useful for the support ticket, could you try it this code in the "live" MakeCode and "beta" version to check if there is any difference?
https://makecode.microbit.org/
https://makecode.microbit.org/beta
```ts
input.onGesture(Gesture.Shake, function () {
    basic.showNumber(1)
})
// input.calibrateCompass()
basic.showNumber(0)
basic.forever(function () {
    basic.showNumber(input.acceleration(Dimension.X))
})
```
There was an improvement added to MakeCode live recently (not yet in beta, oddly enough) that should be able to provide better errors for the motion sensors, so it'd be interesting to see if live MakeCode shows an error code when trying to use the motion sensor directly.

Once you have tried that, please open a support ticket with that information, as this would be off topic for the DAPLink community.

**Comment by mathias-arm** [comment]
Thank you @microbit-carlos
